src/pnpq/devices/odl_ozoptics_650ml.py

```diff
@@ -164,11 +165,19 @@ def serial_read(self, retries=10):
                 got_OK = True
             time.sleep(0.05)
             retries -= 1
+
+        if not got_OK:
+            # means command did not complete before soft timeout!
+            raise RuntimeError(
+                "reading from the device did not complete (soft timeout occured!"
+            )
+
+        self.logger.debug("Device response: %s", device_output)
         return device_output
 
     def serial_command(self, serial_cmd, retries=5):
```

Comment:
The `retries` parameter is now unused, I assume it should be removed.
```suggestion
    def serial_command(self, serial_cmd):
```